From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Tue, 18 Nov 2025 19:30:30 +0000
Subject: fix(systemd-networkd): install and enable
 systemd-networkd-resolve-hook.socket

New socket needed for v259~rc1, networkd fails if this is not installed
and enabled

Closes: #1120955
Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1858
---
 modules.d/11systemd-networkd/module-setup.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/11systemd-networkd/module-setup.sh b/modules.d/11systemd-networkd/module-setup.sh
index 0c6bcf6..7aa3466 100755
--- a/modules.d/11systemd-networkd/module-setup.sh
+++ b/modules.d/11systemd-networkd/module-setup.sh
@@ -53,6 +53,7 @@ install() {
         "$systemdnetwork"/99-default.link \
         "$systemdsystemunitdir"/systemd-networkd.service \
         "$systemdsystemunitdir"/systemd-networkd.socket \
+        "$systemdsystemunitdir"/systemd-networkd-resolve-hook.socket \
         "$systemdsystemunitdir"/systemd-network-generator.service \
         "$systemdsystemunitdir"/systemd-networkd-wait-online.service \
         "$systemdsystemunitdir"/systemd-networkd-wait-online@.service \
@@ -72,6 +73,7 @@ install() {
     for i in \
         systemd-networkd.service \
         systemd-networkd.socket \
+        systemd-networkd-resolve-hook.socket \
         systemd-network-generator.service \
         systemd-networkd-wait-online.service; do
         $SYSTEMCTL -q --root "$initdir" enable "$i"
@@ -87,6 +89,8 @@ install() {
             "$systemdsystemconfdir/systemd-networkd.service.d/*.conf" \
             "$systemdsystemconfdir"/systemd-networkd.socket \
             "$systemdsystemconfdir/systemd-networkd.socket.d/*.conf" \
+            "$systemdsystemconfdir"/systemd-networkd-resolve-hook.socket \
+            "$systemdsystemconfdir/systemd-networkd-resolve-hook.socket.d/*.conf" \
             "$systemdsystemconfdir"/systemd-network-generator.service \
             "$systemdsystemconfdir/systemd-network-generator.service.d/*.conf" \
             "$systemdsystemconfdir"/systemd-networkd-wait-online.service \
